common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -622,14 +626,6 @@ private static boolean hasUnsafe0() {
         }
     }
 
-    private static long arrayBaseOffset0() {
-        if (!hasUnsafe()) {
-            return -1;
-        }
-
-        return PlatformDependent0.arrayBaseOffset();
```

Comment:
We should only delegate to PlatformDepend0.arrayBaseOffset() if Unsafe is present. So maybe cache there but only access if unsafe is available. WDYT ?